hypothesis-python/src/hypothesis/internal/conjecture/data.py

```diff
@@ -1371,80 +1115,35 @@ def _draw_bounded_integer(
         lower: int,
         upper: int,
         *,
-        center: Optional[int] = None,
-        forced: Optional[int] = None,
-        fake_forced: bool = False,
-        _vary_effective_size: bool = True,
+        vary_size: bool = True,
     ) -> int:
         assert lower <= upper
-        assert forced is None or lower <= forced <= upper
         assert self._cd is not None
-        if lower == upper:
-            # Write a value even when this is trivial so that when a bound depends
-            # on other values we don't suddenly disappear when the gap shrinks to
-            # zero - if that happens then often the data stream becomes misaligned
-            # and we fail to shrink in cases where we really should be able to.
-            self._cd.draw_bits(1, forced=0)
-            return int(lower)
-
-        if center is None:
-            center = lower
-        center = min(max(center, lower), upper)
-
-        if center == upper:
-            above = False
-        elif center == lower:
-            above = True
-        else:
-            force_above = None if forced is None else forced < center
-            above = not self._cd.draw_bits(
-                1, forced=force_above, fake_forced=fake_forced
-            )
+        assert self._cd._random is not None
 
-        if above:
-            gap = upper - center
-        else:
-            gap = center - lower
-
-        assert gap > 0
-
-        bits = gap.bit_length()
-        probe = gap + 1
+        if lower == upper:
+            return lower
 
-        if (
-            bits > 24
-            and _vary_effective_size
-            and self.draw_boolean(
-                7 / 8, forced=None if forced is None else False, fake_forced=fake_forced
-            )
-        ):
-            # For large ranges, we combine the uniform random distribution from draw_bits
+        bits = (upper - lower).bit_length()
+        if bits > 24 and vary_size and self._cd._random.random() < 7 / 8:
+            # For large ranges, we combine the uniform random distribution
             # with a weighting scheme with moderate chance.  Cutoff at 2 ** 24 so that our
             # choice of unicode characters is uniform but the 32bit distribution is not.
```

Comment:
Since it's very easy to draw random integers separately for unicode characters now, we might want to revisit this distribution in a future PR.